tests/main/views/test_services.py

```diff
@@ -1022,14 +1023,21 @@ def test_should_update_service_with_valid_statuses(self, index_service):
                         {'updated_by': 'joeblogs'}),
                     content_type='application/json'
                 )
-
+                service = Service.query.filter(
+                    Service.service_id == self.service_id
+                ).one()
                 assert response.status_code == 200
                 data = json.loads(response.get_data())
                 assert status == data['services']['status']
-                if status == 'disabled':
-                    assert index_service.called is False
+                if status in ('disabled', 'enabled'):
```

Comment:
How come we are adding the `enabled` case here?

Reply:
Ah that method only only gets called on publish. 
https://github.com/alphagov/digitalmarketplace-api/blob/master/app/service_utils.py#L134

Reply:
I think that test should probably be split out into 2 but that's a bit out of scope of this PR

Reply:
I don't really like the logic in that view tbf.
It should probably be more like:
```
    if prior_status != status:

        # If it's being unpublished, delete it from the search api.
        if prior_status == 'published':
            delete_service_from_index(service)
        elif status == 'published':
            # If it's being published, index in the search api.
            index_service(service)
        else:
            pass
```

Reply:
Explicit is better than implicit sort of thing

Reply:
https://github.com/alphagov/digitalmarketplace-api/blob/master/app/main/views/services.py#L329

Reply:
Saying that, it hasn't been changed for 2 and a half years, so... If it ain't broke.